File: 067-二进制求和.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        // 0 + 0 = 0, 0 + 1 = 1, 1 + 0 = 1, 1 + 1 = 10
        string sum = "";
        int carry = 0;//进位
        int i = a.size() - 1, j = b.size() - 1;
        while(i >= 0 || j >= 0 || carry == 1)
        {
            if(i >= 0)
                carry += a[i--] - '0';
            if(j >= 0)
                carry += b[j--] - '0';
            sum = char(carry % 2 + '0') + sum;
            carry /= 2;
            
        }
        return sum;
    }
};

File: Java/067-二进制求和.java
class Solution {
    public String addBinary(String a, String b) {
        // 0 + 0 = 0, 0 + 1 = 1, 1 + 0 = 1, 1 + 1 = 10
        String sum = "";
        int carry = 0;// 进位
        int i = a.length() - 1, j = b.length() - 1;
        while (i >= 0 || j >= 0 || carry == 1) {
            if (i >= 0)
                carry += a.charAt(i--) - '0';
            if (j >= 0)
                carry += b.charAt(j--) - '0';
            sum = sum + carry % 2;
            carry /= 2;

        }
        return new StringBuffer(sum).reverse().toString();
    }
}